techmatters/webchat
# Integrate reCAPTCHA v2 checkbox

## Description

- This pull request adds support for reCAPTCHA. The following changes have been made:
- `react-google-recaptcha` library has been added to integrate reCAPTCHA v2 checkbox challenge.
- `enableRecaptcha` flag has been added to the configuration with flag turned on for AS and CA.
- `PreEngagementForm` uses `enableRecaptcha` flag to run recaptcha validation logic.
- This [PR](https://github.com/techmatters/infrastructure-config/pull/193) adds lambda to validate the reCAPTCHA user session

### Checklist
- [n/a] New tests added



### Related Issues
Fixes # https://tech-matters.atlassian.net/browse/CHI-1888

### Verification steps
- `npm ci` and `npm run build:dev` locally, 
- Check that recaptcha appears and user is validated
- Check `devtools > network tab` to ensure successful call to [recaptchaVerify](https://github.com/techmatters/infrastructure-config/pull/193) lambda
- Also, check that no changes or oddities to the PreEngagementForm if `enableRecaptcha` by manually setting to false.
- Screencast - [captcha flow](https://user-images.githubusercontent.com/102122005/236297023-45989046-e972-4956-a23b-868b268ae957.webm)
